### DISCUSSION TITLE
Single log messages are passed to the destination along with aggregated message with grouping-by() parser

### DISCUSSION THREAD

**Discussion by aruaam** [syslog-ng/syslog-ng#4063] [Q&A] [answered]
# syslog-ng
## Version of syslog-ng
syslog-ng 3 (3.37.1)
Config version: 3.35
Installer-Version: 3.37.1
Revision: 3.37.1-1
Compile-Date: May 30 2022 13:00:31
Module-Directory: /usr/lib/syslog-ng/3.37
Module-Path: /usr/lib/syslog-ng/3.37
Include-Path: /usr/share/syslog-ng/include
Available-Modules: appmodel,basicfuncs,sdjournal,linux-kmsg-format,pseudofile,http,cef,afprog,afuser,pacctformat,afsql,json-plugin,system-source,rate-limit-filter,dbparser,csvparser,syslogformat,regexp-parser,tags-parser,disk-buffer,azure-auth-header,kvformat,confgen,timestamp,affile,afmongodb,hook-commands,cryptofuncs,afsocket
Enable-Debug: off
Enable-GProf: off
Enable-Memtrace: off
Enable-IPv6: on
Enable-Spoof-Source: on
Enable-TCP-Wrapper: on
Enable-Linux-Caps: on
Enable-Systemd: on

## Platform
Ubuntu 20.04.4 LTS

## Debug bundle

# Issue
## Failure
When matched message added to the context it is also sent to the destination. After timeout expiration aggregated message posted too. 

## Steps to reproduce
1) Send several messages to syslog-ng from localhost (or any other remote host):
logger -d --rfc3164 -n 127.0.0.1 -P 28282 This is a test.
2) Each message posted separately to the destination telegram channel.
3) After timeout of 10 seconds aggregated message posted as well.

## Configuration
```
@version: 3.37
@include "scl.conf"

# Global options.

options { chain_hostnames(off); flush_lines(0); use_dns(no); use_fqdn(no);
          dns_cache(no); owner("root"); group("adm"); perm(0640);
          stats_freq(0); bad_hostname("^gconfd$");
};

source s_test { network(ip(0.0.0.0) port(28282) transport("udp")); };

destination d_telegram {
   telegram(
    bot-id("bot-id") chat-id("chat-id"));
};

parser p_groupby {
          grouping-by(
                scope(host)
                key("${HOST}")
                timeout(10)
                aggregate()
        );
};

log {
        source (s_test);
        parser (p_groupby);
        destination (d_telegram);
};

###
# Include all config files in /etc/syslog-ng/conf.d/
###
@include "/etc/syslog-ng/conf.d/*.conf"

```
## Input and output logs (if possible)
Ubuntu shell:
Jun 27 15:13:26 127.0.0.1 root: This is a local test.
Jun 27 15:13:27 127.0.0.1 root: This is a local test.
Jun 27 15:13:27 127.0.0.1 root: This is a local test.

Four messages in the Telegram, last one is generated after 10 seconds of timeout:
![image](https://user-images.githubusercontent.com/82602092/175974565-bbf3abc5-a726-43d8-834f-f0df4d5603c9.png)

**Comment by MrAnno** [comment, marked as answer]
Hi,

This is intended behavior.

If you want to discard individual messages, a recently introduced option can be used:

```
grouping-by(inject-mode(aggregate-only) ...)
```
This inject mode will drop individual messages that make up the correlation
context and would only yield the aggregate messages (e.g. the results of
the correlation).

The option is available on our master branch, it has not been released yet.

----

The same can be mimicked by adding a tag to the aggregated message and using a filter to drop everything else.
For example:

```
 grouping-by(
        # ...
        aggregate(
          tags(".tmp.telegram")
        )
        timeout(10)
    );


log {
        source (s_test);
        parser (p_groupby);
        filter { tags(".tmp.telegram"); };
        destination (d_telegram);
};
```

**Comment by aruaam** [comment]
Hi @MrAnno , 

Thanks a lot, it's clear now.
